msdm/domains/gridworld/mdp.py

```diff
@@ -52,17 +52,17 @@ def __init__(self,
                         walls.add(s)
         states.append(TERMINALSTATE)
         actions = [
-            {'dx': 0, 'dy': 0},
-            {'dx': 1, 'dy': 0},
-            {'dx': -1, 'dy': 0},
-            {'dy': 1, 'dx': 0},
-            {'dy': -1, 'dx': 0}
+            frozendict({'dx': 0, 'dy': 0}),
+            frozendict({'dx': 1, 'dy': 0}),
+            frozendict({'dx': -1, 'dy': 0}),
+            frozendict({'dy': 1, 'dx': 0}),
+            frozendict({'dy': -1, 'dx': 0})
         ]
-        self._actions = sorted(actions, key=dictToStr)
-        self._states = sorted(states, key=dictToStr)
-        self._initStates = sorted(list(initStates), key=dictToStr)
-        self._absorbingStates = sorted(list(absorbingStates), key=dictToStr)
-        self._walls = sorted(list(walls), key=dictToStr)
+        self._actions = sorted(actions, key=self.hash_action)
```

Comment:
Since these are now hashable, does `sorted(actions)` work without the `key`?

Reply:
sadly no, frozendict doesn't have comparisons implemented and i couldn't think of a quick fix